pkg/azuredx/resource_handler.go

```diff
@@ -252,3 +260,27 @@ func respondWithError(rw http.ResponseWriter, code int, message string, err erro
 		rw.WriteHeader(http.StatusInternalServerError)
 	}
 }
+
+// sanitizeClusterUri ensures the URI does not contain a query or fragment part
+func sanitizeClusterUri(clusterUri string) (string, error) {
+	// check for trailing question mark before parsing
+	if strings.HasSuffix(clusterUri, "?") {
+		return "", errors.New("clusterUri contains invalid query characters")
```

Comment:
nit - is it worth making this error more specific "clusterUri contains invalid question mark" or something along those lines to differentiate from the error below?

Reply:
Usually you'd be 100% right here, but in this case it actually makes more sense (IMO) to be unspecific, so as not to give potential attackers too much feedback about what they're doing wrong in the API response, and thus making it easier for them. We can see the line the stack trace has come from on our side, which should be good enough for debugging purposes.